[TensorExpr] Fix lowering for aten::div.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48355 [TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead.
* #48354 [TensorExpr] IREval: fix div for Half dtype.
* **#48329 [TensorExpr] Fix lowering for aten::div.**
* #48326 [TensorExpr] Fix aten::atan2 lowering and disable aten::pow lowering on CPU.

Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

Commit:
[TensorExpr] Fix lowering for aten::div.

[ghstack-poisoned]

eellison commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -780,7 +780,7 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
       return computeTwoOperand(
           "aten_div", v, [](const ExprHandle& lhs, const ExprHandle& rhs) {
-            return boolToInteger(lhs) / boolToInteger(rhs);
+            return promoteIntegerToFloat(lhs) / promoteIntegerToFloat(rhs);
```
`torch.rand(4) / torch.tensor([True])`
> tensor([0.5537, 0.8641, 0.2935, 0.0947])
`torch.tensor([True]) / torch.tensor([True])`
> tensor([1.])
I think we need to update promoteIntegerToFloat to use `includeBool = True`

eellison reviewed (approved):
LGTM, one small update to make

Edit: nvm, looks good, was looking at old code in updatte suggestion

Commit:
Update on "[TensorExpr] Fix lowering for aten::div."


Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Fix lowering for aten::div."


Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Fix lowering for aten::div."


Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Fix lowering for aten::div."


Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Fix lowering for aten::div."


Differential Revision: [D25130750](https://our.internmc.facebook.com/intern/diff/D25130750)

[ghstack-poisoned]
